File: backend/codebridge-teams-service/src/main/java/com/codebridge/teams/service/TeamService.java
package com.codebridge.teams.service;

import com.codebridge.teams.dto.TeamDto;
import com.codebridge.teams.dto.TeamMemberDto;
import com.codebridge.teams.dto.TeamRequest;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.ArrayList;
import java.util.List;

/**
 * Service for managing team operations.
 * Handles business logic for team CRUD operations and member management.
 */
@Service
@RequiredArgsConstructor
@Slf4j
public class TeamService {

    /**
     * Get all teams.
     *
     * @return List of teams
     */
    @Transactional(readOnly = true)
    public List<TeamDto> getAllTeams() {
        // Placeholder implementation - replace with actual repository call
        log.info("Getting all teams");
        return new ArrayList<>();
    }

    /**
     * Get teams by organization ID.
     *
     * @param organizationId Organization ID
     * @return List of teams
     */
    @Transactional(readOnly = true)
    public List<TeamDto> getTeamsByOrganization(Long organizationId) {
        // Placeholder implementation - replace with actual repository call
        log.info("Getting teams for organization ID: {}", organizationId);
        return new ArrayList<>();
    }

    /**
     * Get team by ID.
     *
     * @param id Team ID
     * @return Team details
     */
    @Transactional(readOnly = true)
    public TeamDto getTeamById(Long id) {
        // Placeholder implementation - replace with actual repository call
        log.info("Getting team by ID: {}", id);
        return new TeamDto();
    }

    /**
     * Create a new team.
     *
     * @param request Team creation request
     * @return Created team details
     */
    @Transactional
    public TeamDto createTeam(TeamRequest request) {
        // Placeholder implementation - replace with actual repository call
        log.info("Creating new team: {}", request.getName());
        return new TeamDto();
    }

    /**
     * Update a team.
     *
     * @param id Team ID
     * @param request Team update request
     * @return Updated team details
     */
    @Transactional
    public TeamDto updateTeam(Long id, TeamRequest request) {
        // Placeholder implementation - replace with actual repository call
        log.info("Updating team with ID: {}", id);
        return new TeamDto();
    }

    /**
     * Delete a team.
     *
     * @param id Team ID
     */
    @Transactional
    public void deleteTeam(Long id) {
        // Placeholder implementation - replace with actual repository call
        log.info("Deleting team with ID: {}", id);
    }

    /**
     * Get all members of a team.
     *
     * @param id Team ID
     * @return List of team members
     */
    @Transactional(readOnly = true)
    public List<TeamMemberDto> getTeamMembers(Long id) {
        // Placeholder implementation - replace with actual repository call
        log.info("Getting members for team ID: {}", id);
        return new ArrayList<>();
    }

    /**
     * Add a user to a team.
     *
     * @param id Team ID
     * @param userId User ID
     * @param role Role in the team
     */
    @Transactional
    public void addTeamMember(Long id, Long userId, String role) {
        // Placeholder implementation - replace with actual repository call
        log.info("Adding user {} to team {} with role {}", userId, id, role);
    }

    /**
     * Update a team member's role.
     *
     * @param id Team ID
     * @param userId User ID
     * @param role New role
     */
    @Transactional
    public void updateTeamMemberRole(Long id, Long userId, String role) {
        // Placeholder implementation - replace with actual repository call
        log.info("Updating role for user {} in team {} to {}", userId, id, role);
    }

    /**
     * Remove a user from a team.
     *
     * @param id Team ID
     * @param userId User ID
     */
    @Transactional
    public void removeTeamMember(Long id, Long userId) {
        // Placeholder implementation - replace with actual repository call
        log.info("Removing user {} from team {}", userId, id);
    }
}

File: frontend/src/app/core/services/team.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { BaseApiService } from './base-api.service';
import { Team, TeamMember, TeamRequest } from '../models';

@Injectable({
  providedIn: 'root'
})
export class TeamService extends BaseApiService {
  protected override readonly baseUrl = '/teams';

  /**
   * Get all teams
   */
  getTeams(): Observable<Team[]> {
    return this.get<Team[]>(`${this.baseUrl}`);
  }

  /**
   * Get team by ID
   */
  getTeam(teamId: string): Observable<Team> {
    return this.get<Team>(`${this.baseUrl}/${teamId}`);
  }

  /**
   * Create a new team
   */
  createTeam(request: TeamRequest): Observable<Team> {
    return this.post<Team>(`${this.baseUrl}`, request);
  }

  /**
   * Update team
   */
  updateTeam(teamId: string, team: Partial<Team>): Observable<Team> {
    return this.put<Team>(`${this.baseUrl}/${teamId}`, team);
  }

  /**
   * Delete team
   */
  deleteTeam(teamId: string): Observable<void> {
    return this.delete<void>(`${this.baseUrl}/${teamId}`);
  }

  /**
   * Get team members
   */
  getTeamMembers(teamId: string): Observable<TeamMember[]> {
    return this.get<TeamMember[]>(`${this.baseUrl}/${teamId}/members`);
  }

  /**
   * Add team member
   */
  addTeamMember(teamId: string, member: TeamMember): Observable<TeamMember> {
    return this.post<TeamMember>(`${this.baseUrl}/${teamId}/members`, member);
  }

  /**
   * Remove team member
   */
  removeTeamMember(teamId: string, memberId: string): Observable<void> {
    return this.delete<void>(`${this.baseUrl}/${teamId}/members/${memberId}`);
  }
}
